### ISSUE TITLE
Add API to make autoUpdater install from existing file

### ISSUE THREAD

**Issue by juturu** [electron/electron#4603]
- Electron version: 0.36.0
- Operating system: MAC

I want to download update files from my app (due to authenticated end point) and use autoUpdater module to install the downloaded updates. Is this possible? or Any other pointers in installing the update without these modules will also be of help?

I have this mechanism working for squirrel.windows and curious if i can reuse the same on MAC.

**Comment by zcbenz** [comment]
It is not possible on OS X as far as I know, the implementations of auto updater are completely different on OS X and Windows. If you really want this I would suggest posting an issue at https://github.com/Squirrel/Squirrel.Mac.

**Comment by juturu** [comment]
Thanks @zcbenz , I have posted this on Squirrel.Mac (see [this](https://github.com/Squirrel/Squirrel.Mac/issues/166)). Based on the response to squirrel.mac post, it seems autoupdater setFeedUrl method can be updated to set Authorization header on NSURLRequest to support this. Can this be done?

And is it possible to remove the the limitation of calling setFeedUrl only once to be able to update AuthorizationHeader?

**Comment by juturu** [comment]
@zcbenz can you please help here?

**Comment by zcbenz** [comment]
I'm reopening this since it is possible with Squirrel.Mac, we need to add a new API for this, and it would not be very easy.

**Comment by juturu** [comment]
Thanks @zcbenz. When can we can expect new API addition for this?

**Comment by zcbenz** [comment]
I don't have plan to do this recently, a safe guess would be a few months later.

**Comment by sam-jay** [comment]
+1 for setting custom headers on autoupdater requests

**Comment by miniak** [comment]
https://github.com/electron/electron/pull/5879

**Comment by MehediH** [comment]
@zcbenz Sorry for bumping this year old thread but do you possibly have any update on this?

**Comment by tonyganch** [comment]
@miniak @juturu Is this issue still relevant, or should it be closed?

**Comment by codebytere** [comment]
Given that this issue has been present for years with no discernible movement, and that this isn't currently on our roadmap, i'm going to go ahead and label this a `wontfix`. However, should someone determine a workable solution we would be very open to accepting a PR!
